pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepoManager.java

```diff
@@ -388,8 +401,66 @@ private void addExtraFiles(File m2Repo) {
         }
     }
 
+    /**
+     * Generates the offline manifest text file, which contains all the dependencies, including third party, non-RH
+     * dependencies. First, it adds the build artifacts for all the builds. Then, it executes a query to PNC, requesting
+     * the third party dependencies for each build. Finally, the method generates a file, containing version paths and
+     * checksums, where we can get it, for all build and dependency attributes. The offline manifest file in combination
+     * with the offliner tool (https://repo1.maven.org/maven2/com/redhat/red/offliner/offliner) are used to download all
+     * the attributes to a local maven repository.
+     */
+
     @Override
     public void close() {
         buildInfoCollector.close();
     }
+
+    private void generateOfflinerManifest() {
+        log.info("Will generate the Offline manifest");
+        RepositoryData result = new RepositoryData();
+        List<ArtifactWrapper> artifactsToListRaw = new ArrayList<>();
+        for (PncBuild build : builds.values()) {
+            artifactsToListRaw.addAll(build.getBuiltArtifacts());
+            // TODO: Add filter, basing on the targetRepository.repositoryType, when
+            // https://projects.engineering.redhat.com/browse/NCL-6079 is done
+            buildInfoCollector.addDependencies(build, "");
+            artifactsToListRaw.addAll(build.getDependencyArtifacts());
+            log.debug("Dependencies for build {}: {}" + build.getId(), build.getDependencyArtifacts().size());
+        }
+        log.debug("Number of collected artifacts for the Offline manifest: {}", artifactsToListRaw.size());
+
+        artifactsToListRaw.removeIf(artifact -> isArtifactExcluded(artifact.getGapv()));
+        log.debug("Number of collected artifacts after exclusion: {}", artifactsToListRaw.size());
+        Map<String, ArtifactWrapper> artifactsToList = new HashMap<>();
+        for (ArtifactWrapper artifact : artifactsToListRaw) {
+            artifactsToList.put(artifact.getGapv(), artifact);
+        }
+        log.info("Number of collected artifacts without duplicates: {}", artifactsToList.size());
+
+        String filename = releasePath + getGenerationData().getOfflinerManifest();
+        try (PrintWriter file = new PrintWriter(filename)){
```

Comment:
Hi, set the charset here: `new PrintWriter(filename, , StandardCharsets.UTF_8.name())`

Reply:
Also, you're missing a space before the '{' causing the build to fail when it checks the formatting. Alternatively, run `mvn formatter:format` and commit any changes.